core/jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts

```diff
@@ -151,53 +152,80 @@ export class EnvironmentDetailComponent implements OnInit {
-
-        this.cache.set(id, this.service);
-        this.onDataFetched(this.service);
-        this.envoverview.notify(this.service);
-      },
-      err => {
-        this.isLoadingService = false;
-        let errorMessage = this.messageservice.errorMessage(err, "serviceDetail");
-        this.toast_pop('error', 'Oops!', errorMessage)
+
+  fetchService(id: string){
+      this.isLoadingService = true;
+          this.subscription = this.http.get('/jazz/services/'+id).subscribe(
+            response => {
+              this.service.accounts=env_internal.urls.accounts;
+                    this.service.regions=env_internal.urls.regions;
+                  this.service=response.data.data;
+                  if(environment.envName=='oss')this.service=response.data;
+                  this.isFunction= this.service.type === "function";
+                  this.getAssets();
+                  this.setTabs();
+                  this.cache.set(id, this.service);
+                  this.onDataFetched(this.service);
+                  this.envoverview.notify(this.service);
+              },
+              err => {
+                  this.isLoadingService = false;
+                  let errorMessage = this.messageservice.errorMessage(err,"serviceDetail");
+                  this.toast_pop('error', 'Oops!', errorMessage)
 
       }
     )
   };
 
-  testService(type) {
-    switch (type) {
-      case 'api':
-        let swaggerFile = '/' + this.service.domain + '/' + this.service.name + '/' + this.envSelected + '/swagger.json';
-        return window.open(environment.urls['swagger_editor'] + '/?url=' + environment['api_doc_name'] + swaggerFile);
-      case 'website' :
-        if (this.endpoint_env != (undefined || '')) {
-          window.open(this.endpoint_env);
-        }
-        break;
-      case 'function' :
-
-      case 'lamda' :
-        this.setSidebar('try-service');
-        break;
+  setTabs() {
+    if (this.service.serviceType === 'api' || this.service.type === 'api') {
+      this.tabData = ['overview', 'deployments', 'assets', 'metrics', 'code quality', 'logs'];
+    } else if (this.service.serviceType === 'function' || this.service.type === 'function') {
+      this.tabData = ['overview', 'deployments', 'assets', 'metrics', 'code quality', 'logs'];
+    } else if (this.service.serviceType === 'website' || this.service.type === 'website') {
+      this.tabData = ['overview', 'deployments', 'assets', 'metrics'];
     }
   }
 
-  setSidebar(sidebarValue?) {
-    this.sidebar = sidebarValue;
+  getAssets() {
+    this.http.post('/jazz/assets/search', {
+      service: this.service.service || this.service.name,
+      domain: this.service.domain,
+      environment: this.envSelected,
+      limit: undefined
+    }).subscribe((assetsResponse) => {
+      this.assets = assetsResponse.data;
+      this.service.assets = this.assets;
+    }, (err) => {
+      this.toast_pop('error', 'Oops!', 'Swagger File Not Found.');
```

Comment:
Update the message to "Failed to load swagger file."

Reply:
addressed